pantheon-bundle/src/test/java/com/redhat/pantheon/servlet/assembly/AssemblyJsonServletTest.java

```diff
@@ -0,0 +1,141 @@
+package com.redhat.pantheon.servlet.assembly;
+
+import com.redhat.pantheon.model.assembly.Assembly;
+import com.redhat.pantheon.model.module.Module;
+import com.redhat.pantheon.servlet.ServletUtils;
+import com.redhat.pantheon.servlet.assembly.AssemblyJsonServlet;
+import com.redhat.pantheon.servlet.module.ModuleJsonServlet;
+import org.apache.sling.testing.mock.sling.ResourceResolverType;
+import org.apache.sling.testing.mock.sling.junit5.SlingContext;
+import org.apache.sling.testing.mock.sling.junit5.SlingContextExtension;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.EnabledIf;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import javax.jcr.RepositoryException;
+import javax.jcr.query.Query;
+import java.util.Locale;
+import java.util.Map;
+
+import static com.google.common.collect.Maps.newHashMap;
+import static com.redhat.pantheon.util.TestUtils.registerMockAdapter;
+import static javax.servlet.http.HttpServletResponse.SC_OK;
+import static org.junit.jupiter.api.Assertions.*;
+
+@ExtendWith({SlingContextExtension.class})
+class AssemblyJsonServletTest {
+
+    SlingContext slingContext = new SlingContext(ResourceResolverType.JCR_OAK);
+    String testHTML = "<!DOCTYPE html> <html lang=\"en\"> <head><title>test title</title></head> <body " +
+            "class=\"article\"><h1>test title</h1> </header> </body> </html>";
+
+    @Test
+    void getQueryNoParams() {
+        // Given
+        AssemblyJsonServlet servlet = new AssemblyJsonServlet();
+
+        // When
+        String query = servlet.getQuery(slingContext.request());
+
+        // Then
+        // make sure queries don't throw exceptions when executed against the JCR repository
+        assertDoesNotThrow(() -> slingContext.resourceResolver().queryResources(query, Query.JCR_SQL2));
+    }
+
+    @Test
+    void getQuery() {
+        // Given
+        AssemblyJsonServlet servlet = new AssemblyJsonServlet();
+        Map<String, Object> map = newHashMap();
+        map.put("locale", ServletUtils.toLanguageTag(Locale.US));
+        map.put("module_id", "jcr:uuid");
```

Comment:
i wonder if we should call it something more generic so that both module api and assembly api can share the same name. Hydra team maps this field to id field in solr. The fact is this is assembly_id not module_id. Both are uuid of a node. we could call it "uuid", which will make the mapping more straightforward for hydra team.

Reply:
That is a good idea. So, rename (assembly, module)_uuid to just uuid in the response only or also in the url as well?

Reply:
I would rename in both the response body. I am not sure which url you refer to. 
    the url to module api looks like this: https://<HOST>/api/module/variant.json/b537ef3c-5c7d-4280-91ce-e7e818e6cc11
    the url to assembly api looks like this: https://<HOST>/api/assembly.json/bc3ae91d-fbf1-4dc5-9bec-edfaa03d29fa

Reply:
if you do decide to rename line 51 to uuid instead of "module_id", it would be great if you can add this to VariantJsonServlet.java:
       // TODO: remove module_uuid after Hydra team releases UNIFIED-6570
        variantMap.put("module_uuid", variant_uuid);
       // add the following to the map for hdyra to consume
        variantMap.put("uuid", variant_uuid);

Reply:
Ah.. ok, my bad regarding the URI. I am fine with renaming in assembly but module response is being used by  teams other than Hydra. So, it could introduce bugs. Hence, I am hesitant and not in favor of making change  to module response in the scope of this JIRA/PR. Maybe we can have another JIRA for that where other teams can be notified?

Reply:
yes, that works for me.

Reply:
I have updated the test case accordingly. Will raise another JIRA for common/generic name for uuid